core/pkg/fixhandler/yamlhandler.go

```diff
@@ -0,0 +1,302 @@
+package fixhandler
+
+import (
+	"bufio"
+	"bytes"
+	"container/list"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/mikefarah/yq/v4/pkg/yqlib"
+
+	"gopkg.in/yaml.v3"
+)
+
+func constructDecodedYaml(filepath string) *[]yaml.Node {
+	file, err := ioutil.ReadFile(filepath)
+	if err != nil {
+		logger.L().Fatal("Cannot read file")
+	}
+	fileReader := bytes.NewReader(file)
+	dec := yaml.NewDecoder(fileReader)
+
+	nodes := make([]yaml.Node, 0)
+	for {
+		var node yaml.Node
+		err = dec.Decode(&node)
+		nodes = append(nodes, node)
+
+		// break the loop in case of EOF
+		if errors.Is(err, io.EOF) {
+			break
+		}
+		if err != nil {
+			panic(err)
+		}
+	}
+
+	return &nodes
+}
+
+func constructFixedYamlNodes(filePath, yamlExpression string) (*[]yaml.Node, error) {
+	preferences := yqlib.ConfiguredYamlPreferences
+	preferences.EvaluateTogether = true
+	decoder := yqlib.NewYamlDecoder(preferences)
+
+	var allDocuments = list.New()
+	reader, err := constructNewReader(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	fileDocuments, err := readDocuments(reader, filePath, 0, decoder)
+	if err != nil {
+		return nil, err
+	}
+	allDocuments.PushBackList(fileDocuments)
+
+	allAtOnceEvaluator := yqlib.NewAllAtOnceEvaluator()
+
+	fixedCandidateNodes, err := allAtOnceEvaluator.EvaluateCandidateNodes(yamlExpression, allDocuments)
+
+	if err != nil {
+		logger.L().Fatal(fmt.Sprintf("Error fixing YAML, %v", err.Error()))
+	}
+
+	fixedNodes := make([]yaml.Node, 0)
+	var fixedNode *yaml.Node
+	for fixedCandidateNode := fixedCandidateNodes.Front(); fixedCandidateNode != nil; fixedCandidateNode = fixedCandidateNode.Next() {
+		fixedNode = fixedCandidateNode.Value.(*yqlib.CandidateNode).Node
+		fixedNodes = append(fixedNodes, *fixedNode)
+	}
+
+	return &fixedNodes, nil
+}
+
+func constructDFSOrder(node *yaml.Node) *[]NodeInfo {
+	dfsOrder := make([]NodeInfo, 0)
+	constructDFSOrderHelper(node, nil, &dfsOrder, 0)
```

Comment:
issue: the helper function is in a whole different file

We should place the helper function nearby, otherwise it’s especially difficult to navigate during code review

Reply:
Agree!! I'll place them nearby each other.